### PR TITLE
Develop/drgn 10805 create rng driver

### PR DESCRIPTION
**Pull request by PavelVPV**

### CODE REVIEW

File: drivers/entropy/Kconfig.nrf5 [hunk 1]
```diff
@@ -61,10 +76,10 @@ config ENTROPY_NRF5_ISR_THRESHOLD
 	  buffer goes below this number hardware entropy generation will be
 	  started.
 
+endif #ENTROPY_NRF5_RNG_DEFAULT
 
 config ENTROPY_NRF5_PRI
 	int "RNG interrupt priority"
-	depends on ENTROPY_NRF5_RNG
```
Comment (omkar3141) [inline drivers/entropy/Kconfig.nrf5:67]: I think this dependency should not be removed. Since driver will never be used if ENTROPY_NRF5_RNG is not selected.

File: drivers/entropy/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,20 @@
+if (CONFIG_ENTROPY_ESP32_RNG OR
```
Comment (omkar3141) [inline drivers/entropy/CMakeLists.txt]: This `if ... endif` looks unnecessary. The `zephyr_library_sources_ifdef()` does the same thing.

Comment (PavelVPV) [reply drivers/entropy/CMakeLists.txt]: The thing is that zephyr_library() function forces to have at least one source file for the library.
Removing zephyr_library() will cause the following error:

> CMake Error at ../../../cmake/extensions.cmake:360 (add_library):
>   Cannot find source file:
> 
>     entropy_nrf5.c

This is the bad hack to avoid the problem, but still have ENTROPY_HAS_DRIVER enabled. I've asked about how to solve this.

File: drivers/entropy/Kconfig.nrf5 (hunk 1 above)
Comment (PavelVPV) [reply drivers/entropy/Kconfig.nrf5:67]: It is under if ENTROPY_NRF5_RNG/endif, so the dependency is redundant

File: drivers/clock_control/Kconfig.nrf5 [hunk 3]
```diff
@@ -5,29 +5,27 @@
 # SPDX-License-Identifier: Apache-2.0
 
 menuconfig CLOCK_CONTROL_NRF5
-        bool "NRF5 Clock controller support"
```
Comment (rugeGerritsen) [inline drivers/clock_control/Kconfig.nrf5]: I don't think you should change this

File: samples/drivers/entropy/src/main.c [hunk 4]
```diff
@@ -25,7 +25,7 @@ void main(void)
 
 	while (1) {
 #define BUFFER_LENGTH 10
-		u8_t buffer[BUFFER_LENGTH] = {0};
+		u8_t buffer[BUFFER_LENGTH] = { 0 };
```
Comment (rugeGerritsen) [inline samples/drivers/entropy/src/main.c]: Please revert the changes in this file

Comment (PavelVPV) [reply samples/drivers/entropy/src/main.c]: Done

File: drivers/entropy/CMakeLists.txt (hunk 2 above)
Comment (PavelVPV) [reply drivers/entropy/CMakeLists.txt]: Provided better solutions for this problem

File: drivers/entropy/CMakeLists.txt [hunk 5]
```diff
@@ -1,10 +1,8 @@
...
-
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_ESP32_RNG          entropy_esp32.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_RNGA          entropy_mcux_rnga.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_MCUX_TRNG          entropy_mcux_trng.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_NRF5_RNG           entropy_nrf5.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_SAM_RNG            entropy_sam.c)
-zephyr_library_sources_ifdef(CONFIG_ENTROPY_STM32_RNG          entropy_stm32.c)
-zephyr_library_sources_ifdef(CONFIG_FAKE_ENTROPY_NATIVE_POSIX  fake_entropy_native_posix.c)
-zephyr_library_sources_ifdef(CONFIG_USERSPACE                  entropy_handlers.c)
+zephyr_sources_ifdef(CONFIG_ENTROPY_ESP32_RNG          entropy_esp32.c)
```
Comment (rugeGerritsen) [inline drivers/entropy/CMakeLists.txt:1]: What is the difference between from zephyr_library_sources_ifdef and zephyr_sources_ifdef?

Comment (PavelVPV) [reply drivers/entropy/CMakeLists.txt:1]: zephyr_library() creates a library using cmake function add_library() and then adds sources to it by calling target_sources(${LIBRARY_NAME} PRIVATE ${source_file}) through zephyr_library_sources... function. 

zephyr_sources...() adds sources to "zephyr" target like this: target_sources(zephyr PRIVATE ${source_file})
